c7n/resources/ecs.py

```diff
@@ -247,6 +247,90 @@ class ServiceTaskDefinitionFilter(RelatedTaskDefinitionFilter):
     """
 
 
+@Service.action_registry.register('modify')
+class UpdateService(BaseAction):
+    """Action to update service
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: no-public-ips-services
+                resource: ecs-service
+                filters:
+                  - 'networkConfiguration.awsvpcConfiguration.assignPublicIp': 'ENABLED'
+                actions:
+                  - type: modify
+                    update:
+                      networkConfiguration:
+                        awsvpcConfiguration:
+                          assignPublicIp: DISABLED
+    """
+
+    schema = type_schema('modify',
+        update={
+            'cluster': {'type': 'string'},
+            'service': {'type': 'string'},
+            'desiredCount': {'type': 'integer'},
+            'taskDefinition': {'type': 'string'},
+            'deploymentConfiguration': {
+                'type': 'object',
+                'properties': {
+                    'maximumPercent': {'type': 'integer'},
+                    'minimumHealthyPercent': {'type': 'integer'},
+                }
+            },
+            'networkConfiguration': {
+                'type': 'object',
+                'properties': {
+                    'awsvpcConfiguration': {
+                        'type': 'object',
+                        'properties': {
+                            'subnets': {
```

Comment:
also may want to include `subnets_from`, `securityGroups_from` to support modifying subnets/sg's from a list (see implementation in notify as an example (to_from, cc_from, etc.))

Reply:
could you clarify what you mean by that.. these are already lists, and they these are straight pass throughs to api parameters.

Reply:
Also, for better or worse, I made the **type_schema** closely match the actual AWS API call. The only thing I changed in the data structure is adding the **update**  level (ie the one that used to be **modify**).

Reply:
I meant lists from a external source like s3 or a file, but for now looks fine-- this may be a future enhancement